docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+        """
+        ...
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        context: os.PathLike[str],
+        tag: str,
+        *,
+        dockerfile_string: str,
+        output_file: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile_string : str
+            A Dockerfile-formatted string containing the information necessary
+            to build the desired image.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
+            program will be written to. If None, the method will not output to
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile-formatted string at context/dockerfile_loc
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        context,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        output_file=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None:
+            dockerfile_build = True
+        elif dockerfile_string is not None:
+            dockerfile_build = False
+        else:
+            return ValueError("Image.build: either dockerfile or " +
+                              "dockerfile_string must be defined.")
+
+        if output_file is not None:
+            run_stdout = output_file
+        else:
+            run_stdout = PIPE
+
+        context_str = os.fspath(context)
+        cmd = [
+            "docker",
+            "build",
+            f"--network={network}",
+            context_str,
+            "-t",
+            tag
+        ]
+
+        if no_cache:
+            cmd += ["--no-cache"]
+
+        if dockerfile_build:
+            cmd += [f"--file={os.fspath(dockerfile)}"]
+            stdin_input = None
+        else:
+            cmd += ["-f-"]
+            stdin_input = dockerfile_string
+
+        run(
+            cmd,
+            text=True,
+            stdout=run_stdout,  # type: ignore
+            stderr=STDOUT,
```

Comment:
>`stderr=STDOUT`

I don't think this is the behavior that we want. I can't think of a situation where we'd prefer to redirect STDERR messages to STDOUT.

```suggestion
```